api/server/router/local/image.go

```diff
@@ -284,6 +294,18 @@ func (s *router) postBuild(ctx context.Context, w http.ResponseWriter, r *http.R
 	w.Header().Set("Content-Type", "application/json")
 
 	version := httputils.VersionFromContext(ctx)
+	output := ioutils.NewWriteFlusher(w)
+	sf := streamformatter.NewJSONStreamFormatter()
+	errf := func(err error) error {
+		// Do not write the error in the http output if it's still empty.
+		// This prevents from writing a 200(OK) when there is an interal error.
+		if !output.Flushed() {
+			return err
+		}
+		w.Write(sf.FormatError(errors.New(utils.GetErrorMessage(err))))
+		return nil
```

Comment:
@LK4D4 for the case when the HTTP header wasn't written yet, it returns the error, which is then translated by httputil.WriteError (called from api/server/server.go) to an HTTP status code.